### PR TITLE
Rename sync versions with correct context

### PR DESCRIPTION
**Pull request by ChintanRaval**
rename a given `data_load_execution`'s sync versions as below:
- `this_sync_version` to `last_sync_version` (which was and is used to store the previous sync version)
- `next_sync_version` to `sync_version` (which was and is used to store the current sync version)

### CODE REVIEW

File: modules/DestinationTableManager.py [hunk 1]
```diff
@@ -114,22 +114,16 @@ def upsert_table(self, schema_name, source_table_name, target_table_name, column
         primary_key_column_list = ','.join(map(str, primary_key_column_array))
 
         sql_builder = io.StringIO()
-        sql_builder.write(f"INSERT INTO {schema_name}.{target_table_name} ({column_list})")
-        sql_builder.write(os.linesep)
-        sql_builder.write(f" SELECT {column_list} FROM {schema_name}.{source_table_name}")
-        sql_builder.write(os.linesep)
+        sql_builder.write(f"INSERT INTO {schema_name}.{target_table_name} ({column_list}) \n")
```
Comment (seanbudd) [inline modules/DestinationTableManager.py]: should this be changed to f-strings to keep with convention

File: modules/data_load_tracking/DataLoadExecution.py [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
     id = Column(Integer, primary_key=True)
+    correlation_id = Column(UUID(as_uuid=True), nullable=True)
     model_name = Column(String(250), nullable=False)
+    status = Column(String(25), nullable=False)
+    last_sync_version = Column(BigInteger, nullable=False)
+    sync_version = Column(BigInteger, nullable=False)
     is_full_refresh = Column(Boolean, nullable=False)
-    this_sync_version = Column(BigInteger, nullable=False)
-    next_sync_version = Column(BigInteger, nullable=False)
+    full_refresh_reason = Column(String(100), nullable=False)
```
Comment (seanbudd) [inline modules/data_load_tracking/DataLoadExecution.py]: are we still changing this to codes?

File: modules/shared/Utils.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import importlib
+
+
+def create_type_instance(type_name):
```
Comment (seanbudd) [inline modules/shared/Utils.py]: what's this used for?

File: modules/DestinationTableManager.py (hunk 1 above)
Comment (ChintanRaval) [reply modules/DestinationTableManager.py]: tried. but keeping it the way it is with `.format()` and `io.StringIO` as it looks simpler.

File: modules/data_load_tracking/DataLoadExecution.py (hunk 2 above)
Comment (ChintanRaval) [reply modules/data_load_tracking/DataLoadExecution.py]: so want to.. but not in this PR.

File: modules/shared/Utils.py (hunk 3 above)
Comment (ChintanRaval) [reply modules/shared/Utils.py]: nothing actually.. wasn't used.. and isn't used. there's commented code that _was_ using this.. so just keeping it isolated in a separate module `Utils` until we figure out if its ever needed.

**Pull request merged**